crates/ursa-proxy/src/core/handler.rs

```diff
@@ -83,3 +172,20 @@ pub async fn proxy_pass<C: Cache>(
     };
     StreamBody::new(ReaderStream::new(reader)).into_response()
 }
+
+async fn serve_file_handler(
+    OriginalUri(uri): OriginalUri,
+    Extension(root): Extension<String>,
+) -> Response {
+    let mut file_path = PathBuf::from_str(root.as_str()).unwrap();
+    file_path.push(uri.path().trim_start_matches('/'));
+    let path_str = file_path.as_os_str().to_str().unwrap();
```

Comment:
Yes, this is possible. Can this happen with the current set up or does NGINX have some measures in place?

Reply:
I think it would be better to:
1. resolve the path using `use resolve_path::PathResolveExt`
2. double check if it does not contain .. after resolve.
3. double check if the final path to read from starts with the directory path from config using [`starts_with`](https://doc.rust-lang.org/stable/std/path/struct.PathBuf.html#method.starts_with)

Reply:
@kckeiks nvm my previous comment, can we use `tower_http:: services::ServeDir` and eliminate this function? 

https://github.com/tokio-rs/axum/blob/main/examples/static-file-server/src/main.rs#L44

it already has protection in place
https://docs.rs/tower-http/latest/src/tower_http/services/fs/serve_dir/mod.rs.html#36